index.rst

```diff
@@ -2,28 +2,137 @@
+
+The diagram below summarizes the design of the system.
+
+.. figure:: /_static/ToO_diagram.png
+   :name: fig-too-diagram
+   :target: ../_images/ToO_diagram.png
+   :alt: ToO automatic response system design.
+
+   Design of the system to autonomously handle ToO with the Scheduler CSC.
+
+The service in charge of reading/receiving ToO alerts from external sources is the ``ToO Alert Producer``, shown in the :ref:`diagram above <fig-too-diagram>`.
+This service will also be in charge of certifying/validating which ToO events the Scheduler will consider.
+
+Furthermore, the ``ToO Alert Producer`` can be designed in such a way to support multiple ``ToO Alert Stream`` sources.
+
+Manual Response With the ScriptQueue
+------------------------------------
+
+As mentioned above, in order for the Scheduler CSC to be able to respond to a certain type of ToO, it needs to be pre-configured with a setup capable of processing the incoming data stream and acting accordingly.
+This process should work on those cases where we know what to expect and are capable of planning ahead of time (e.g. Gravitational-wave events that got pre-approved by the project).
+
+Nevertheless, it might be that an important unplanned event occurs and requires immediate response, for which the Scheduler CSC won't be capable of handling.
+
+For those situations, a list of observations can be generated and added to the ScriptQueue.
+Since the Scheduler and the ScriptQueue are designed to seamlessly interoperate, it is possible the manually add a series of observations to the queue without worrying about the Scheduler.
+
+In these cases, the ScriptQueue will execute the observations, the Scheduler will wait for the execution to complete and then resume operations.
+
+It is worth noticing that this use-case might be a good application of a service like `SkyPortal`_, which allows users to interactively select and process ToO events and generate sequences of observations.
+
+.. _SkyPortal: https://skyportal.io
+
+The main draw-back of this approach is that we lose a lot of the responsiveness provided by the Scheduler to the overall observatory conditions.
+For example, the Scheduler is able to take into account the wind speed and direction when deciding which targets to observe, a condition that can change pretty drastically in a short timescale.
+It is also possible to account for things like the telescope azimuth phase-wrap, thus avoiding long slews to unwrap it.
+All these conditions, and many more, are not easily taken into account if the sequence of observations is pre-computed externally and sent to the ScriptQueue.
+
+Therefore, although we acknowledge that this is a viable option for responding to unplanned ToOs, we expect to be able to respond to most of them autonomously through the Scheduler CSC.
```

Comment:
This is an interesting idea. I didn't read it that way so much, but I'm coming in with a different viewpoint. 
For example, I would think that maybe what we would do is take some of the capabilities from SkyPortal (choosing which ToO events are scientifically 'interesting' to follow up on given a limited amount of time available, identifying the limiting magnitude we would want to try to reach for each bandpass given the information in the ToO, and defining the sky-area to observe for the ToO) and then feed that into the EFD or ToO Alert system, and then using that as part of the input to the FBS. 
Which, I guess is what Michael was suggesting, but was something I was taking somewhat for granted as an option.

Reply:
Ok, I see what you mean.. I should have mentioned something in section 4.1.1, but yeah, my idea is that a service like that could be used as the "ToO Alert Stream", which would allow us to augment the data with some user interaction.